core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/BaseResourceSearchRequest.java

```diff
@@ -85,6 +88,27 @@ protected final Expression prepareQuery(RepositoryContext context) {
 		return queryBuilder.build();
 	}
 	
+	/**
+	 * Configures security filters to allow access to certain resources only. This method is no-op if the given {@link User} is an administrator or has read access to everything. 
+	 * 
+	 * @param queryBuilder - the query builder to append the clauses to
+	 * @param user - the user who's permissions will be applied to this resource search request
+	 */
+	protected final void addSecurityFilter(ExpressionBuilder queryBuilder, User user) {
+		if (!user.isAdministrator() && !user.hasPermission(Permission.requireAll(Permission.OPERATION_BROWSE, Permission.ALL))) {
+			Set<String> permittedResourceIds = user.getPermissions().stream().flatMap(p -> p.getResources().stream()).collect(Collectors.toSet());
```

Comment:
Does having any sort of permission on a resource imply `BROWSE` rights? With the currently used operations (edit, import, export, version, promote task, classify), it feels so.

Reply:
Yes, if you have at least one permission with other than the `browse` scope that includes the `browse` scope as well. For example, you would not be able to edit a resource without reading it first if this wasn't true.